-------------------------------------------------------------------------------
Test set: org.apache.hadoop.ozone.TestSecureOzoneCluster
-------------------------------------------------------------------------------
Tests run: 10, Failures: 1, Errors: 2, Skipped: 0, Time elapsed: 114.755 s <<< FAILURE! - in org.apache.hadoop.ozone.TestSecureOzoneCluster
testSecureOmInitSuccess(org.apache.hadoop.ozone.TestSecureOzoneCluster)  Time elapsed: 80.021 s  <<< ERROR!
java.lang.Exception: test timed out after 80000 milliseconds
	at java.lang.Thread.sleep(Native Method)
	at org.apache.hadoop.util.ThreadUtil.sleepAtLeastIgnoreInterrupts(ThreadUtil.java:45)
	at org.apache.hadoop.hdds.utils.RetriableTask.call(RetriableTask.java:64)
	at org.apache.hadoop.ozone.om.ScmClient.getScmInfo(ScmClient.java:118)
	at org.apache.hadoop.ozone.om.init.FullUnsecureInitializer.initialize(FullUnsecureInitializer.java:36)
	at org.apache.hadoop.ozone.om.init.FullSecureInitializer.initialize(FullSecureInitializer.java:33)
	at org.apache.hadoop.ozone.om.init.OzoneManagerStorageInitializer.run(OzoneManagerStorageInitializer.java:64)
	at org.apache.hadoop.ozone.TestSecureOzoneCluster.initializeOmStorage(TestSecureOzoneCluster.java:882)
	at org.apache.hadoop.ozone.TestSecureOzoneCluster.testSecureOmInitSuccess(TestSecureOzoneCluster.java:813)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$StatementThread.run(FailOnTimeout.java:74)

testSecureOmInitializationSuccess(org.apache.hadoop.ozone.TestSecureOzoneCluster)  Time elapsed: 3.939 s  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at org.apache.hadoop.ozone.TestSecureOzoneCluster.testSecureOmInitializationSuccess(TestSecureOzoneCluster.java:422)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$StatementThread.run(FailOnTimeout.java:74)

testSecureOMInitializationFailure(org.apache.hadoop.ozone.TestSecureOzoneCluster)  Time elapsed: 3.627 s  <<< ERROR!
org.apache.hadoop.security.authentication.client.AuthenticationException: Login failed for non-existent-user@EXAMPLE.com with keytab: /workdir/hadoop-ozone/integration-test/target/test-dir/TestSecureOzoneCluster/om.keytab
	at org.apache.hadoop.ozone.OzoneSecurityUtil.login(OzoneSecurityUtil.java:91)
	at org.apache.hadoop.ozone.om.OzoneManager.<init>(OzoneManager.java:310)
	at org.apache.hadoop.ozone.om.OzoneManager.createOm(OzoneManager.java:753)
	at org.apache.hadoop.ozone.TestSecureOzoneCluster.lambda$testSecureOMInitializationFailure$5(TestSecureOzoneCluster.java:400)
	at org.apache.hadoop.test.LambdaTestUtils.intercept(LambdaTestUtils.java:491)
	at org.apache.hadoop.test.LambdaTestUtils.intercept(LambdaTestUtils.java:377)
	at org.apache.hadoop.test.LambdaTestUtils.intercept(LambdaTestUtils.java:446)
	at org.apache.hadoop.ozone.TestSecureOzoneCluster.testCommonKerberosFailures(TestSecureOzoneCluster.java:371)
	at org.apache.hadoop.ozone.TestSecureOzoneCluster.testSecureOMInitializationFailure(TestSecureOzoneCluster.java:400)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$StatementThread.run(FailOnTimeout.java:74)
Caused by: org.apache.hadoop.security.KerberosAuthException: failure to login: for principal: non-existent-user@EXAMPLE.com from keytab /workdir/hadoop-ozone/integration-test/target/test-dir/TestSecureOzoneCluster/om.keytab javax.security.auth.login.LoginException: Unable to obtain password from user

	at org.apache.hadoop.security.UserGroupInformation.doSubjectLogin(UserGroupInformation.java:1847)
	at org.apache.hadoop.security.UserGroupInformation.loginUserFromKeytabAndReturnUGI(UserGroupInformation.java:1215)
	at org.apache.hadoop.security.UserGroupInformation.loginUserFromKeytab(UserGroupInformation.java:1008)
	at org.apache.hadoop.security.SecurityUtil.login(SecurityUtil.java:315)
	at org.apache.hadoop.ozone.OzoneSecurityUtil.login(OzoneSecurityUtil.java:87)
	... 19 more
Caused by: javax.security.auth.login.LoginException: Unable to obtain password from user

	at com.sun.security.auth.module.Krb5LoginModule.promptForPass(Krb5LoginModule.java:897)
	at com.sun.security.auth.module.Krb5LoginModule.attemptAuthentication(Krb5LoginModule.java:760)
	at com.sun.security.auth.module.Krb5LoginModule.login(Krb5LoginModule.java:617)
	at sun.reflect.GeneratedMethodAccessor19.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at javax.security.auth.login.LoginContext.invoke(LoginContext.java:755)
	at javax.security.auth.login.LoginContext.access$000(LoginContext.java:195)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:682)
	at javax.security.auth.login.LoginContext$4.run(LoginContext.java:680)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:680)
	at javax.security.auth.login.LoginContext.login(LoginContext.java:587)
	at org.apache.hadoop.security.UserGroupInformation$HadoopLoginContext.login(UserGroupInformation.java:1926)
	at org.apache.hadoop.security.UserGroupInformation.doSubjectLogin(UserGroupInformation.java:1837)
	... 23 more

